src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -244,6 +244,7 @@ public FetchCommand fetch_() {
             public boolean shallow;
             public Integer timeout;
             public boolean tags = true;
+            public Integer depth = 1;
```

Comment:
will your code work if it will be null? Do you want handle all nulls instead of using primitives?
If it not a designed use case i will prefer to skip tri-state fields.